cr17/nodes/cr17/navigator.py

```diff
@@ -27,6 +27,81 @@
 REACHED_GOAL_TOPIC = rospy.get_param("topics/reached_goal", "reached_goal")
 OBSTACLES_TOPIC = rospy.get_param("topics/obstacles","/obstacle_centroids")
 
+def at_goal(robot_pose, goal):
+    '''
+    Given a robot_pose and a goal coordinate, this node determines if robot is at the goal 
+    '''
+    # calc distance 
+    dist = math.sqrt((goal.x - robot_pose.position.x)**2 + (goal.y - robot_pose.position.y)**2)
+    at_goal = True if dist <= GOAL_THRESH else False
+    return at_goal
+
+def calc_goal_force(nav_goal, robot_pose):
+    '''
+    given a goal point and a robot pose, calculate and return x and y components of goal force
+    '''
+    FIELD_SPREAD = 10.0 # radius around goal where pfield is scaled
+    ALPHA = 1.0
+    # get distance between goal and robot
+    dist = math.sqrt((nav_goal.x - robot_pose.position.x)**2 + (nav_goal.y - robot_pose.position.y)**2)
+    # get angle to goal
+    angle_to_goal = math.atan2(nav_goal.y - robot_pose.position.y, nav_goal.x - robot_pose.position.x)
+    # get force angle
+    force_angle = wrap_angle(angle_to_goal)
+    # math the components
+    if dist < GOAL_THRESH:
+        d_x = 0
+        d_y = 0
+    elif (GOAL_THRESH <= dist) and (dist <= FIELD_SPREAD + GOAL_THRESH):
+        d_x = ALPHA * (dist - GOAL_THRESH) * math.cos(force_angle)
+        d_y = ALPHA * (dist - GOAL_THRESH) * math.sin(force_angle)
+    else: #dist > (FIELD_SPREAD + GOAL_THRESH)
+        d_x = ALPHA * FIELD_SPREAD * math.cos(force_angle)
+        d_y = ALPHA * FIELD_SPREAD * math.sin(force_angle)
+
+    return [d_x, d_y]
```

Comment:
There was a place in a different function where it normalized the vector, but that was changed to use tuples. Changed in 2b4de4372834138b7cde16f7b3bd395bf3d75540